### DISCUSSION TITLE
Clarification for Power calibration feature

### DISCUSSION THREAD

**Discussion by testter2** [openthread/openthread#9862] [Q&A] [answered]
Hello,

I'm interested in new Power calibration feature and wants to get more info about it.

previously power was set using otPlatRadioSetTransmitPower API before transmission. so, why new APIs are needed for power calibration feature ?

Is it for tuning power of ack for received packet of specified channel ? if yes then I think otPlatRadioSetChannelTargetPower  do the job instead of otPlatRadioSetTransmitPower , right ?

I'm also wondering about the need of other 3 APIs otPlatRadioAddCalibratedPower, otPlatRadioClearCalibratedPowers, otPlatRadioGetRawPowerSetting specially when platform only has single power table?

Thanks

**Comment by jwhui** [comment]
The `otPlatRadioAddCalibratedPower()` API provides more flexibility to specify exactly how a given platform configures its transmit power. As noted in the comments, some platforms may include both a radio and a FEM, which can be individually controlled to achieve a desired output power.

**Comment by zhanglongxia** [comment, marked as answer]
There is a limitation in FCC, the out-of-band spurious emissions at the band edge is 20dB below the fundamental emission level. In order to meet this requirement, we set the output power of edge channels smaller than other channels. This allows Thread to use maximum power to send frames while meeting regulations. So we added the `otPlatRadioSetChannelTargetPower ()` for developers to set different output powers for different channels.

The API `otPlatRadioAddCalibratedPower()`, `otPlatRadioClearCalibratedPowers()` and `otPlatRadioGetRawPowerSetting ()` are used for setting the power calibration table. If the output power of your radio chip is very accurate, there is no need to use these three APIs.

**Comment by testter2** [comment]
Thank you for the detailed answers @zhanglongxia @jwhui
